src/lib/convert.ts

```diff
@@ -46,10 +85,10 @@ class Transpiler {
             variable = 'forloop.counter0';
           } else if (path.original === '@key') {
             // TODO: support "@../key"
-            variable = 'key';
+            // variable = 'key';
+            variable = 'forloop.counter0';
```

Comment:
In hbs/js, you would only use `@key` for objects (although it works for arrays as well), so we either assume we have to switch to object-for, or we throw an error if we feel the wrong each-version is used.